src/legacy/core_plugins/embeddable_api/public/triggers/attach_action.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IRegistry, Trigger } from '../types';
+
+export function attachAction(
+  triggerRegistry: IRegistry<Trigger>,
+  { triggerId, actionId }: { triggerId: string; actionId: string }
+) {
+  const trigger = triggerRegistry.get(triggerId);
+  if (!trigger) {
+    throw new Error(`No trigger with is ${triggerId} exists`);
+  }
+
+  if (!trigger.actionIds.find(id => id === actionId)) {
+    trigger.actionIds.push(actionId);
```

Comment:
we don't perform any check if specific action can even work with this trigger ? 
so we can attach any action to any trigger, only when trigger is emitted we would check if action is compatible, and even there we check if they are compattible with embeddable, not with triggerContext ?

Reply:
The isCompatible checks are done at time of execution when the triggerContext and embeddable instance are available.  I couldn't add in here a check because I don't have the context yet.  It's left up to the developer to know when it makes sense to attach a given action to a given trigger Id.

The interesting thing to work through will be what happens when we don't have the dynamic trigger context available and we have to check if something will be compatible to show the user, for phase 2.  I think that is where trigger output spec's will come into play.  iI will be static information about the expected trigger context that we can use to determine compatibility.  We will still have at that time the embeddable context though, since the end user will be setting it up per embeddable instance.  So at this point, we have none of that information.

Open for thoughts on how to improve this, but for now, yes, the developer should take care to only attach actions to trigger ids that makes sense.  A developer could attach the APPLY_FILTER_ACTION to the SHOW_CONTEXT_MENU_TRIGGER and get no warnings, but once the user opens the context menu, and clicks "Apply filter", nothing would happen because triggerContext would be empty at that time.

Reply:
action could define the type of actual trigger context it expects